java-squid/src/main/java/org/sonar/java/resolve/Resolve.java

```diff
@@ -377,17 +377,22 @@ public Resolution findMethod(Env env, String name, List<JavaType> argTypes, List
   }
 
   public Resolution findMethod(Env env, JavaType site, String name, List<JavaType> argTypes) {
-    return findMethod(env, site, name, argTypes, ImmutableList.<JavaType>of(), false);
+    return findMethod(env, site, site, name, argTypes, ImmutableList.<JavaType>of());
```

Comment:
Maybe you could have let the `false` parameter to avoid the initial call to the new method `findMethod`